From ab6324ff26144b9f36458b80d8b3c91f80ba79ea Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Mon, 26 Dec 2016 15:54:57 +0200
Subject: [PATCH 1190/2241] dts: cp110: sata: add sata-port property to sata
 node

Generic AHCI drivers allows to parse sub-notes under SATA
device-tree node and treat those sub-nodes as SATA ports,
this can be useful for parsing PHYs for each port.

This patch has no functional change, but it allows to describe
the SATA nodes in a more precise way
(giving information regarding functional and dis-functional ports).
In addition, sata-port nodes will be used for future comphy support

Change-Id: Ifed5e8412e7e9ab3cc4752d952036a95dc589eb0
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35025
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts   |  7 +++++++
 arch/arm64/boot/dts/marvell/armada-7040-db-router.dts |  7 +++++++
 arch/arm64/boot/dts/marvell/armada-70x0-customer.dts  |  9 +++++++++
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts      | 14 ++++++++++++++
 arch/arm64/boot/dts/marvell/armada-8040-db-router.dts | 14 ++++++++++++++
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts     | 14 ++++++++++++++
 arch/arm64/boot/dts/marvell/armada-80x0-customer.dts  | 18 ++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi       | 11 +++++++++++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi         | 11 +++++++++++
 9 files changed, 105 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
index ff2e823..534eaa0 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
@@ -87,6 +87,13 @@
 			};
 			sata: sata@540000 {
 				status = "okay";
+
+				sata-port@0 {
+					status = "disabled";
+				};
+				sata-port@1 {
+					status = "okay";
+				};
 			};
 			usb3h0: usb3@500000 {
 				status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
index a78f081..32e05c9 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
@@ -87,6 +87,13 @@
 			};
 			sata: sata@540000 {
 				status = "okay";
+
+				sata-port@0 {
+					status = "disabled";
+				};
+				sata-port@1 {
+					status = "okay";
+				};
 			};
 			usb3h0: usb3@500000 {
 				status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
index ff6fa53..dd148bf 100644
--- a/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
+++ b/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
@@ -128,6 +128,15 @@
 			};
 			sata: sata@540000 {
 				status = "disabled";
+
+				sata-port@0 {
+					/* UPDATE */
+					status = "disabled";
+				};
+				sata-port@1 {
+					/* UPDATE */
+					status = "disabled";
+				};
 			};
 			usb3h0: usb3@500000 {
 				status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
index e0106bd..5fb0a0c 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
@@ -77,6 +77,13 @@
 		config-space {
 			sata@540000 {
 				status = "okay";
+
+				sata-port@0 {
+					satus = "okay";
+				};
+				sata-port@1 {
+					status = "okay";
+				};
 			};
 			/* usb3h0 is USB2 only (UTMI) */
 			usb3h0: usb3@500000 {
@@ -149,6 +156,13 @@
 			};
 			sata@540000 {
 				status = "okay";
+
+				sata-port@0 {
+					status = "okay";
+				};
+				sata-port@1 {
+					status = "okay";
+				};
 			};
 			mdio@12a200 {
 				#address-cells = <1>;
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
index c9ca277..121a1b7 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
@@ -109,6 +109,13 @@
 		config-space {
 			sata@540000 {
 				status = "okay";
+
+				sata-port@0 {
+					satus = "disabled";
+				};
+				sata-port@1 {
+					status = "okay";
+				};
 			};
 			usb3h0: usb3@500000 {
 				status = "okay";
@@ -177,6 +184,13 @@
 			};
 			sata@540000 {
 				status = "okay";
+
+				sata-port@0 {
+					satus = "disabled";
+				};
+				sata-port@1 {
+					status = "okay";
+				};
 			};
 			mdio@12a200 {
 				#address-cells = <1>;
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index b453c36..ee3fd20 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -113,6 +113,13 @@
 			};
 			sata@540000 {
 				status = "okay";
+
+				sata-port@0 {
+					status = "disabled";
+				};
+				sata-port@1 {
+					status = "okay";
+				};
 			};
 			gop {
 				emac0_0: mac0 {
@@ -165,6 +172,13 @@
 			};
 			sata@540000 {
 				status = "okay";
+
+				sata-port@0 {
+					status = "okay";
+				};
+				sata-port@1 {
+					status = "okay";
+				};
 			};
 			gop {
 				emac0_1: mac0 {
diff --git a/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
index 5c8cf07..e178aa1 100644
--- a/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
+++ b/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
@@ -137,6 +137,15 @@
 		config-space {
 			sata@540000 {
 				status = "disabled";
+
+				sata-port@0 {
+					/* UPDATE */
+					status = "disabled";
+				};
+				sata-port@1 {
+					/* UPDATE */
+					status = "disabled";
+				};
 			};
 			usb3h0: usb3@500000 {
 				status = "disabled";
@@ -243,6 +252,15 @@
 			};
 			sata@540000 {
 				status = "disabled";
+
+				sata-port@0 {
+					/* UPDATE */
+					status = "disabled";
+				};
+				sata-port@1 {
+					/* UPDATE */
+					status = "disabled";
+				};
 			};
 			mdio@12a200 {
 				/* UPDATE */
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index 8c58c67..ae3c925 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -72,6 +72,17 @@ sata@540000 {
 	status = "disabled";
 	port_base = <0x10000>;
 	port_offset = <0x10000>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	sata-port@0 {
+		reg = <0>;
+		satus = "disabled";
+	};
+	sata-port@1 {
+		reg = <1>;
+		status = "disabled";
+	};
 };
 
 usb3h0_1: usb3@500000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 61a0446..abef02d 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -143,6 +143,17 @@ sata: sata@540000 {
 	status = "disabled";
 	port_base = <0x10000>;
 	port_offset = <0x10000>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	sata-port@0 {
+		reg = <0>;
+		satus = "disabled";
+	};
+	sata-port@1 {
+		reg = <1>;
+		status = "disabled";
+	};
 };
 
 usb3h0: usb3@500000 {
-- 
2.7.4

